### DISCUSSION TITLE
How to use multiple types in record value?

### DISCUSSION THREAD

**Discussion by dhruv-nudge** [sinclairzx81/typebox#837] [Q&A] [answered]
I want to do something like this
```
props: Type.Record(Type.String(), Type.Number() | Type.String() | Type.Boolean() | Type.Date())
```

**Comment by sinclairzx81** [comment, marked as answer]
@dhruv-nudge Hiya, try the following

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxnAL5wBmUEIcA5AAIDOwAdiQDYHBQD0MdBgBGEAB4MAUFJIR2LeMjgBeFMIB0AJQxyoAEwAUaTBvxQOAc0MBKHCYwaAqu2DzDAbSlwf60wDkAVxARDChbLG9fBzMYC3ZrOyifGIAhCAguDAJ2COS-RwARIgxbKQBdGxsZIUwUVTxCYhIAHlqMCCoUAD4Zfn44dvq1BCl+3zgPcQAuOEV4y3LZ+as4AB84dmDQqHW4MUzs9j3imAwAbjGB8iA)

```typescript
import { Type, Static } from '@sinclair/typebox'

const T = Type.Record(Type.String(), Type.Union([
    Type.Number(),
    Type.String(),
    Type.Boolean(),
    Type.Date()
]))

type T = Static<typeof T>

// type T = {
//    [x: string]: string | number | boolean | Date;
// }
```
Hope this helps
S

**Reply by Dhruv-Garg79** [reply]
Hey just like there is Type.Integer, is there anything for Type.Decimal?
